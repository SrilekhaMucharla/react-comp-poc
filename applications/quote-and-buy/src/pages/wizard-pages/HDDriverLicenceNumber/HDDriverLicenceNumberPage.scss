.dln-container {
  h3 {
    @include media-breakpoint-down(sm) {
      font-size: 25px;
    }
  }

  &__error {
    margin-top: 15px;
    margin-bottom: -15px;
  }

  .img-fluid {
    @include media-breakpoint-down(sm) {
      width: 100%;
    }
  }
}
//PAGE SPECIFIC but must be taken down to global level
.footer-button-large + .wizard-navigation {
  .wizard-buttons-footer .btn {
    @include media-breakpoint-up(lg) {
      width: 50%;
    }
  }
  .wizard-navigation-forward {
    margin-top: -7%;
  }
}

.dln__title {
  & > div:nth-child(2) {
    padding-left: 0.5em;
    @include media-breakpoint-up(md) {
      padding-left: 2em;
    }
  }
}

.button-continue {
    width: 300px;
    margin: 0 330px 20px 0;
    padding: 13px 112px 13px 111px;
    border-radius: 6px;
    margin-top: 6%;
    margin-left: 3%;
}
