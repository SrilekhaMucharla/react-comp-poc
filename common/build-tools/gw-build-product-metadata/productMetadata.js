const _ = require('lodash');
const { readJson } = require('fs-extra');
const appPaths = require('gw-build-config-paths');
const { sendJsonRpcRequest } = require('gw-build-jsonrpc');
const fileUtils = require('./util/fileUtils');
const metadataLocalisation = require('./metadataLocalisation');

function toMetadataMap(serverMetadataResult) {
    const metadataByName = _.groupBy(serverMetadataResult, 'name');
    return _.mapValues(metadataByName, (typeMetas) => {
        const allMetadataForName = _.map(typeMetas, 'metadata');
        return _.merge({}, ...allMetadataForName);
    });
}

/**
 * Fetches the metadata for the given endpoint
 *
 * @param {ApplicationEndpoint} endpoint the endpoint which metadata should be fetched
 * @returns {Promise<Object>} the metadata retrieved from the given endpoint
 */
function fetchMetadataForEndpoint(endpoint) {
    return sendJsonRpcRequest(endpoint, 'getMetaData')
        .then((metadata) => ({
            endpoint,
            metadata
        }));
}

/**
 * @typedef { import("gw-build-capabilities-management").ApplicationEndpoint } ApplicationEndpoint
 */

/**
 * @typedef PreprocessMetadata
 * @property {{server: string, service: string}} endpoint the server nickname (e.g. pc, cc, ...)
 * @property {object} metadata the metadata from the server
 */

/**
 * Structures the product metadata
 *
 * @param {PreprocessMetadata} data containing the endpoint from which
 * the metadata came from and the metadata from the server
 * @returns {Object} the metadata retrieved from the given endpoint
 */
function preprocessDtoMetadata({ endpoint, metadata }) {
    return {
        [endpoint.server]: toMetadataMap(metadata.newTypeMeta)
    };
}

/**
 * Structures the question set metadata
 *
 * @param {PreprocessMetadata} data containing the endpoint from which
 * the metadata came from and the metadata from the server
 * @returns {Object} the metadata from the server
 */
function preprocessQuestionSetsMetadata({ metadata }) {
    return metadata.questionSets;
}

/**
 * Combines all the metadatas from all the endpoints into one object
 *
 * @param {Array<Object>} allMetadataResponses containing the metadata from the server
 * @returns {Object} the metadata to save
 */
function mergeMetadata(allMetadataResponses) {
    return allMetadataResponses.reduce((generatedMetadata, serviceMetadata) => {
        return _.merge(generatedMetadata, serviceMetadata);
    }, {});
}

/**
 * Pipeline to format, combine and save the metadata
 *
 * @param {Promise<Object>} metadataPromises response from the server
 * @param {function} preprocessingFunction function to format the metadata from the server
 * @param {string} destinationFile path to save the metadata
 * @returns {Object} the metadata to save
 */
function metadataPipeline(metadataPromises, preprocessingFunction, destinationFile) {
    return metadataPromises
        .then((metadataFromEndpoints) => metadataFromEndpoints.map(preprocessingFunction))
        .then(mergeMetadata)
        .then((allMetadata) => fileUtils.writeFile(allMetadata, destinationFile));
}

/**
 * Process the product metadata
 *
 * @param {Promise<Object>} metadataPromises response from the server
 * @returns {Promise} the metadata to save
 */
function processDTOMetadata(metadataPromises) {
    return metadataPipeline(metadataPromises, preprocessDtoMetadata, appPaths.appProductMetadata);
}

/**
 * Process the question sets metadata
 *
 * @param {Promise<Object>} metadataPromises response from the server
 * @returns {Promise} the metadata to save
 */
function processQuestionSetsMetadata(metadataPromises) {
    return metadataPipeline(metadataPromises, preprocessQuestionSetsMetadata,
        appPaths.appQuestionSetsMetadata);
}

async function buildMetadata() {
    const { capabilities } = await readJson(appPaths.appGeneratedConfig);
    const endpoints = Object.values(capabilities);
    const endpointsWithMetadata = endpoints.filter((endpoint) => !endpoint.skipMetadata);
    const metadataPromises = Promise.all(endpointsWithMetadata.map(fetchMetadataForEndpoint));
    await Promise.all([
        processDTOMetadata(metadataPromises),
        processQuestionSetsMetadata(metadataPromises)
    ]);
    const localeConfig = await metadataLocalisation.fetchLocaleInfo();
    metadataLocalisation.processMetadataDisplayKeys(localeConfig, endpoints);
}

module.exports = {
    buildMetadata
};
