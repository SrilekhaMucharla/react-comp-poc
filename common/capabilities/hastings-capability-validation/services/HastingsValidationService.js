import { getProxiedServiceUrl } from 'gw-portals-url-js';
import { BaseTransportService } from 'gw-portals-transport-js';

export default class HastingsValidationService {
    /**
     * Retrieves LicenseDataDTO on license number.
     *
     * @param {Object} dto LicenseRequestDTO
     * @returns {Promise}
     * @memberof HastingsValidationService
     */
    static validateLicense(dto) {
        return BaseTransportService.send(
            getProxiedServiceUrl('validateLicense'),
            {
                'Content-Type': 'application/json; charset=utf-8'
            },
            {
                jsonrpc: '2.0',
                method: 'validateLicense',
                params: [dto]
            }
        );
    }

    static validateBankAccount(dto) {
        return BaseTransportService.send(
            getProxiedServiceUrl('validateLicense'),
            {
                'Content-Type': 'application/json; charset=utf-8'
            },
            {
                jsonrpc: '2.0',
                method: 'validateBankAccount',
                params: [dto]
            }
        );
    }

    static checkTransactionDetails(dto) {
        return BaseTransportService.send(
            getProxiedServiceUrl('validateLicense'),
            {
                'Content-Type': 'application/json; charset=utf-8'
            },
            {
                jsonrpc: '2.0',
                method: 'checkTransactionDetails',
                params: [dto]
            }
        );
    }

    static checkTransactionDetailsForMC(dto) {
        return BaseTransportService.send(
            getProxiedServiceUrl('validateLicense'),
            {
                'Content-Type': 'application/json; charset=utf-8'
            },
            {
                jsonrpc: '2.0',
                method: 'checkTransactionDetailsForMC',
                params: [dto]
            }
        );
    }
}
