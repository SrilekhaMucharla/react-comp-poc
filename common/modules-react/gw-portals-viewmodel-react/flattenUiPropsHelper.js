/* eslint-disable */
import _ from "lodash";

/**
 *
 * @param {object} uiProps
 */
export function getFlattenedUiPropsContent(uiProps) {
  if (uiProps.contentRepeat) {
    return uiProps;
  }

  if (!_.isArray(uiProps.content)) {
    return uiProps;
  }

  return uiProps.content.flatMap((item) => getFlattenedUiPropsContent(item));
}
