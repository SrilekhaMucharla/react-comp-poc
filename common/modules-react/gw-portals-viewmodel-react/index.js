export { default as ViewModelServiceContext } from './ViewModelServiceContext';
export { default as withViewModelService } from './withViewModelServiceHOC';
export { getFlattenedUiPropsContent } from './flattenUiPropsHelper';
