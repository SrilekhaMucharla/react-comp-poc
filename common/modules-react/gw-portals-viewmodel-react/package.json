{
  "name": "gw-portals-viewmodel-react",
  "description": "Provides API to work with View Model",
  "main": "index.js",
  "version": "11.2.0-db467149eb42",
  "author": "Guidewire",
  "private": true,
  "dependencies": {},
  "peerDependencies": {
    "react": "16.x",
    "lodash": "4.x",
    "prop-types": "15.x"
  }
}
