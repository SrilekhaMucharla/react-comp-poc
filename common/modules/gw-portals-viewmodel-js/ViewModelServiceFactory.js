import InsuranceSuiteMetadataServiceFactory from 'gw-portals-metadata-js';
import DateUtil from 'gw-portals-util-js/DateUtil';
import LocalDateUtil from 'gw-portals-util-js/LocalDateUtil';

import ViewModel from './ViewModel';

import Context from './aspects/Context';
import Requiredness from './aspects/Requiredness';
import Validity from './aspects/Validity';
import SubtreeValidity from './aspects/SubtreeValidity';
import InputCtrlType from './aspects/InputCtrlType';
import AvailableValues from './aspects/AvailableValues';
import SerializedForm from './aspects/SerializedForm';
import RebuildVMFromData from './aspects/RebuildVMFromData';
import DebugInvalidChildProperties from './aspects/DebugInvalidChildProperties';
import TypedValue from './aspects/TypedValue';
import _ValidationFunctions from './aspects/_ValidationFunctions';
import _AddressAutocompleteUtil from './aspects/_AddressAutocompleteUtil';

import { ExpressionLanguageServiceFactory } from './expression-language';
import _PhoneUtil from './util/PhoneUtil';
import DateFormatter from './util/DateFormatter';

function getViewModelService(metadata, translator) {
    const suiteMetadata = InsuranceSuiteMetadataServiceFactory.createSuiteMetadata(metadata);
    const ExpressionLanguageService = ExpressionLanguageServiceFactory.get(
        { insuranceSuiteMetadataService: suiteMetadata, translator }
    );
    // view model initiation which is done in the initial module in Portals (extracted)
    const ViewModelService = ViewModel.get(suiteMetadata);

    const typeValidationMessages = {};
    typeValidationMessages.notANumber = translator('platform.modelvalidation.DTOValidationService.Please enter numbers only');
    typeValidationMessages.notAnInteger = translator('platform.modelvalidation.DTOValidationService.Value entered must not contain decimal values');

    ViewModelService.addAspectFactory(Context.create(ExpressionLanguageService));
    ViewModelService.addAspectFactory(Requiredness.create(ExpressionLanguageService));
    ViewModelService.addAspectFactory(
        Validity.create(ExpressionLanguageService, typeValidationMessages)
    );
    ViewModelService.addAspectFactory(SubtreeValidity);
    ViewModelService.addAspectFactory(InputCtrlType);
    ViewModelService.addAspectFactory(AvailableValues.create(ExpressionLanguageService));
    ViewModelService.addAspectFactory(SerializedForm);
    ViewModelService.addAspectFactory(RebuildVMFromData);
    ViewModelService.addAspectFactory(DebugInvalidChildProperties);
    ViewModelService.addAspectFactory(TypedValue);

    ExpressionLanguageService.getCompilationContext().registerType('edgev10.aspects.validation.ValidationFunctions', _ValidationFunctions());
    ExpressionLanguageService.getCompilationContext().registerType('gw.api.contact.AddressAutocompleteUtil', _AddressAutocompleteUtil());
    ExpressionLanguageService.getCompilationContext().registerType('edgev10.capabilities.locale.util.DateFormatter', DateFormatter);
    ExpressionLanguageService.getCompilationContext().registerType('gw.api.util.DateUtil', DateUtil);
    ExpressionLanguageService.getCompilationContext().registerType('gw.api.util.PhoneUtil', _PhoneUtil());
    ExpressionLanguageService.getCompilationContext().registerType('edgev10.time.LocalDateUtil', LocalDateUtil);

    return {
        create: ViewModelService.create,
        changeContext: ViewModelService.changeContext,
        clone: ViewModelService.clone
    };
}

// EXPORT
export default {
    getViewModelService
};
