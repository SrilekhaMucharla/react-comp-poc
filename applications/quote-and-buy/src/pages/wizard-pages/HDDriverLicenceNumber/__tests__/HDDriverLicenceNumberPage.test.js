import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import _ from 'lodash';

// GW
import { ViewModelServiceContext } from 'gw-portals-viewmodel-react';
import ViewModelServiceFactory from 'gw-portals-viewmodel-js/ViewModelServiceFactory';

// HD
// Full path import is required to apply mock !!
import HastingsValidationService from 'hastings-capability-validation/services/HastingsValidationService';
import HDDriverLicenceNumberPage from '../HDDriverLicenceNumberPage';
import ukDrivingLicense from '../UkDrivingLicenseGenerator';
import productMetadata from '../../../../generated/metadata/product-metadata.json';
import defaultTranslator from '../../__helpers__/testHelper';
import withTranslator from '../../__helpers__/test/withTranslator';
import submission from '../../../../routes/SubmissionVMInitial';

const middlewares = [];
const mockStore = configureStore(middlewares);
const submissionVM = {};
let wrapper;

const driverPath = 'lobData.privateCar.coverables.drivers.children[0]';
const licenceTypeFieldname = 'licenceType';
const licenceTypePath = `${driverPath}.${licenceTypeFieldname}.value.code`;

const updateEvent = {
    target: {
        value: 'DR9AB'
    },
};

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({})
}));

// Init VM
const viewModelService = ViewModelServiceFactory.getViewModelService(
    productMetadata, defaultTranslator
);

const simulateDLNChange = async (dlnInput) => {
    // Simulate entering new value
    await act(async () => {
        dlnInput.at(0)
            .props()
            .onChange(updateEvent);
    });
};

describe('<HDDriverLicenceNumberPage />', () => {
    beforeEach(() => {
        // Initialize mockstore with empty state
        const lastNameFieldName = 'lastName';
        const lastNamePath = `${driverPath}.person.${lastNameFieldName}.value`;
        const genderPath = `${driverPath}.gender.value.code`;

        const driverBornDayPath = `${driverPath}.dateOfBirth.day.value`;
        const driverBornMonthPath = `${driverPath}.dateOfBirth.month.value`;
        const driverBornYearPath = `${driverPath}.dateOfBirth.year.value`;
        const licenceNumberFieldname = 'licenseNumber';
        const licenceNumberPath = `${driverPath}.${licenceNumberFieldname}`;

        // Set default values
        _.set(submissionVM, lastNamePath, 'jones');
        _.set(submissionVM, genderPath, 'F');
        _.set(submissionVM, driverBornYearPath, 1976);
        // 0-11 month notation
        _.set(submissionVM, driverBornMonthPath, 2);
        _.set(submissionVM, driverBornDayPath, 11);
        _.set(submissionVM, licenceTypePath, 'F_FM');

        // valid license
        _.set(submissionVM, `${licenceNumberPath}.aspects.subtreeValid`, true);

        const initialState = {
            wizardState: {
                data: {
                    submissionVM: submissionVM
                },
                app: {
                    step: 1,
                    prevStep: 0,
                    pages: {
                        drivers: {
                            0: { licenceSuccessfulScanned: false, licenceSuccessfulValidated: false },
                        }
                    }
                },
            }
        };

        const store = mockStore(initialState);

        wrapper = mount(withTranslator(
            <ViewModelServiceContext.Provider value={viewModelService}>
                <Provider store={store}>
                    <HDDriverLicenceNumberPage pageId="test-page" />
                </Provider>
            </ViewModelServiceContext.Provider>
        ));
    });

    test('Render component', () => {
        expect(wrapper)
            .toMatchSnapshot();
    });

    test('Don\'t validate DLN on external service', async () => {
        HastingsValidationService.validateLicense = jest.fn(
            () => Promise.resolve({
                result: {
                    drivingLicence: {
                        drivingLicenceNumber: updateEvent.target.value
                    }
                }
            })
        );

        const dln = wrapper.find('#partTwo');
        _.set(submissionVM, licenceTypePath, 'P_PU');

        // Simulate entering new value
        await simulateDLNChange(dln);

        expect(HastingsValidationService.validateLicense)
            .toHaveBeenCalledTimes(0);
    });

    test('Validate DLN on external service', async () => {
        HastingsValidationService.validateLicense = jest.fn(
            () => Promise.resolve({
                result: {
                    drivingLicence: {
                        directiveIndicator: 4,
                        drivingLicenceNumber: 'RNEWQ801019DR9AB',
                        drivingLicenceYears: 2,
                        validFrom: '2017-06-26T23:00:00Z',
                        validTo: '2067-06-23T23:00:00Z',
                        drivingLicenceDate: '2017-06-26T23:00:00Z',
                        dvlaProcessingDate: '2018-04-22T14:03:52Z',
                        dvlaServiceVersion: '1',
                        hubProcessingDate: '2018-04-22T14:03:52Z',
                        hubServiceVersion: '1.0.0.0'
                    },
                    drivingEndorsementsCollection: [
                        {
                            code: 'SP30',
                            convictionDate: '2020-11-11T00:00:00Z',
                            custodialPeriod: 0,
                            disqualPeriod: 'PT0S ',
                            fine: 0,
                            isDisqual: false,
                            noOfPoints: 3,
                            offenceDate: '2020-09-09T23:00:00Z'
                        },
                        {
                            code: 'SP30',
                            convictionDate: '2017-11-11T00:00:00Z',
                            custodialPeriod: 0,
                            disqualEndDate: '2018-11-11T00:00:00Z',
                            disqualPeriod: 'P1Y0M0DT12H30M5S ',
                            disqualStartDate: '2017-11-11T00:00:00Z',
                            fine: 600,
                            isDisqual: true,
                            noOfPoints: 3,
                            rehabSpentDate: '2019-11-11T00:00:00Z'
                        }
                    ],
                }
            })
        );

        // Initialize mockstore with empty state
        const mockSubmissionVM = viewModelService.create(
            submission,
            'pc',
            'edgev10.capabilities.quote.submission.dto.QuoteDataDTO'
        );

        const licenceHeldForFieldname = 'licenceHeldFor';
        const licenceHeldForPath = `${driverPath}.${licenceHeldForFieldname}`;

        const yearLicensedAvailableValues = _.get(mockSubmissionVM, licenceHeldForPath).aspects.availableValues;
        _.set(submissionVM, `${licenceHeldForPath}.aspects.availableValues`, yearLicensedAvailableValues);

        const dln = wrapper.find('#partTwo');

        // Simulate entering new value
        await simulateDLNChange(dln);

        const storeVM = wrapper.find('Provider').props().store.getState().wizardState.data.submissionVM;

        const driver = _.get(storeVM, `${driverPath}`);

        expect(HastingsValidationService.validateLicense)
            .toHaveBeenCalled();

        expect(driver.claimsAndConvictions.anyConvictions.value).toBeTruthy();
        expect(driver.claimsAndConvictions.convictionsCollection[0].drivingBanMonths).toBe(0);
        expect(driver.claimsAndConvictions.convictionsCollection[1].drivingBanMonths).toBe(12);
    });

    test('Failed request to validate DLN', async () => {
        HastingsValidationService.validateLicense = jest.fn(
            () => Promise.resolve({
                result: {
                    errorCode: '603'
                }
            })
        );

        const dln = wrapper.find('#partTwo');

        // Simulate entering new value
        await simulateDLNChange(dln);

        wrapper.update();

        const alert = wrapper.find('HDAlertRefactor');

        expect(alert.text())
            .toBeTruthy();
    });
});

describe('ukDrivingLicenseNumber', () => {
    test('Female driving license', () => {
        const drivingLicenseNumber = ukDrivingLicense('jones', 1976, 3, 11, 'F');

        expect(drivingLicenseNumber)
            .toBe('JONES753116');
    });

    test('Female driving license, date birth December', () => {
        const drivingLicenseNumber = ukDrivingLicense('jones', 1976, 12, 11, 'F');

        expect(drivingLicenseNumber)
            .toBe('JONES762116');
    });

    test('Male driving license', () => {
        const drivingLicenseNumber = ukDrivingLicense('judd', 1959, 7, 13, 'M');

        expect(drivingLicenseNumber)
            .toBe('JUDD9507139');
    });

    test('Throw error on bad input', () => {
        expect(() => { ukDrivingLicense('judd', 1959, 7, 13, 1); })
            .toThrowError();
    });
});
