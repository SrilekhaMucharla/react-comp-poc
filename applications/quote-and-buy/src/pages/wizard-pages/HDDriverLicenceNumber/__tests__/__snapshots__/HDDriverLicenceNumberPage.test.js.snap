// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDDriverLicenceNumberPage /> Render component 1`] = `
<Component>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(HDDriverLicenceNumberPage)
      pageId="test-page"
    >
      <HDDriverLicenceNumberPage
        pageId="test-page"
        setErrorStatusCode={[Function]}
        setNavigation={[Function]}
        setWizardPageState={[Function]}
        submissionVM={
          Object {
            "lobData": Object {
              "privateCar": Object {
                "coverables": Object {
                  "drivers": Object {
                    "children": Array [
                      Object {
                        "dateOfBirth": Object {
                          "day": Object {
                            "value": 11,
                          },
                          "month": Object {
                            "value": 2,
                          },
                          "year": Object {
                            "value": 1976,
                          },
                        },
                        "gender": Object {
                          "value": Object {
                            "code": "F",
                          },
                        },
                        "licenceType": Object {
                          "value": Object {
                            "code": "F_FM",
                          },
                        },
                        "licenseNumber": Object {
                          "aspects": Object {
                            "subtreeValid": true,
                          },
                        },
                        "person": Object {
                          "lastName": Object {
                            "value": "jones",
                          },
                        },
                      },
                    ],
                  },
                },
              },
            },
          }
        }
      >
        <Container
          className="dln-container margin-bottom-lg footer-button-large"
          fluid={false}
        >
          <div
            className="dln-container margin-bottom-lg footer-button-large container"
          >
            <Row
              noGutters={false}
            >
              <div
                className="row"
              >
                <Col>
                  <div
                    className="col"
                  >
                    <HDLabelRefactor
                      Tag="h2"
                      additionalContent={
                        <h3
                          className="d-inline"
                        >
                           (optional)
                        </h3>
                      }
                      adjustImagePosition={true}
                      className="dln__title"
                      icon={
                        <img
                          alt="discount"
                          className="mb-2"
                          src="discount.png"
                        />
                      }
                      iconPosition="r"
                      id="dln-discount-label"
                      text="Tell us your driving licence number and you could get a discount..."
                    >
                      <Row
                        className="label-with-icon align-items-start dln__title"
                        noGutters={false}
                      >
                        <div
                          className="label-with-icon align-items-start dln__title row"
                        >
                          <Col
                            className="pr-0"
                          >
                            <div
                              className="pr-0 col"
                            >
                              <h2
                                className="dln__title"
                                id="dln-discount-label"
                              >
                                Tell us your driving licence number and you could get a discount...
                                <h3
                                  className="d-inline"
                                >
                                   (optional)
                                </h3>
                              </h2>
                            </div>
                          </Col>
                          <Col
                            className="mt-h2  "
                            xs="auto"
                          >
                            <div
                              className="mt-h2   col-auto"
                            >
                              <img
                                alt="discount"
                                className="mb-2"
                                src="discount.png"
                              />
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </HDLabelRefactor>
                    <div
                      className="text-small"
                    >
                      *Based on online quoted premiums from June to August 2021
                    </div>
                    <HDInfoCardRefactor
                      className="margin-top-lg margin-bottom-lg"
                      id="dln-parking-general-info"
                      image="exclamation-icon.svg"
                      isRight={false}
                      notes={null}
                      paragraphs={
                        Array [
                          "We'll be able to access your driving information from the DVLA, so we can  calculate any discounts you qualify for, based on your driving record.",
                          "Your licence must be from England, Wales or Scotland as DVLA data isn't available for licences from Northern Ireland, the Isle of Man or the Channel Islands.",
                        ]
                      }
                      size={null}
                      theme={null}
                      title={null}
                    >
                      <div
                        className="info-card left  margin-top-lg margin-bottom-lg"
                        id="dln-parking-general-info"
                      >
                        <img
                          alt="info-card"
                          className="info-card__icon"
                          src="exclamation-icon.svg"
                        />
                        <p
                          className="info-card__paragraph paragraph_0"
                          key="0"
                        >
                          We'll be able to access your driving information from the DVLA, so we can  calculate any discounts you qualify for, based on your driving record.
                        </p>
                        <p
                          className="info-card__paragraph paragraph_1"
                          key="1"
                        >
                          Your licence must be from England, Wales or Scotland as DVLA data isn't available for licences from Northern Ireland, the Isle of Man or the Channel Islands.
                        </p>
                      </div>
                    </HDInfoCardRefactor>
                    <div
                      className="margin-bottom-lg"
                    >
                      <img
                        alt="DLN"
                        className="img-fluid"
                        src="DLN_Image.png"
                      />
                    </div>
                    <HDLabelRefactor
                      Tag="h3"
                      adjustImagePosition={true}
                      className=""
                      id="dln-enter-dln-label"
                      text="Please enter the next five
characters:"
                    >
                      <h3
                        className=""
                        id="dln-enter-dln-label"
                      >
                        Please enter the next five
characters:
                      </h3>
                    </HDLabelRefactor>
                    <Row
                      noGutters={false}
                    >
                      <div
                        className="row"
                      >
                        <Col
                          className="pr-0 pr-sm-2"
                          md={4}
                          xs={7}
                        >
                          <div
                            className="pr-0 pr-sm-2 col-md-4 col-7"
                          >
                            <ForwardRef
                              className="font-bold dln-container__dln-prefix-input"
                              disabled={true}
                              id="partOne"
                              name="partOne"
                              onBeforeOpen={null}
                              onBlur={null}
                              onCancel={null}
                              onChange={[Function]}
                              onClick={null}
                              onConfirm={null}
                              onDelete={null}
                              onEdit={null}
                              onFocus={null}
                              onSelect={null}
                              type="alphanum"
                              value="JONES753116"
                              webAnalyticsEvent={null}
                            >
                              <ForwardRef
                                allowLeadingZero={false}
                                appendContent={null}
                                appendLabel={null}
                                className="font-bold dln-container__dln-prefix-input"
                                customClassName=""
                                data={null}
                                disabled={true}
                                icon={null}
                                id="partOne"
                                inputMode={null}
                                isInvalidCustom={null}
                                mask={null}
                                maskChar="_"
                                maxLength={null}
                                name="partOne"
                                onBeforeOpen={null}
                                onBlur={null}
                                onCancel={null}
                                onChange={[Function]}
                                onClick={null}
                                onConfirm={null}
                                onDelete={null}
                                onEdit={null}
                                onFocus={null}
                                onKeyPress={[Function]}
                                onSelect={null}
                                path="path.to.input"
                                placeholder={null}
                                preText={null}
                                reset={false}
                                size={null}
                                svgIcon={null}
                                thousandSeprator={false}
                                tickIcon={false}
                                type="alphanum"
                                value="JONES753116"
                                webAnalyticsEvent={null}
                              >
                                <InputGroup
                                  className="font-bold dln-container__dln-prefix-input "
                                  size={null}
                                >
                                  <div
                                    className="font-bold dln-container__dln-prefix-input  input-group"
                                  >
                                    <FormControl
                                      className="hd-text-input__form-control"
                                      data-testid="text-input"
                                      disabled={true}
                                      id="partOne"
                                      inputMode={null}
                                      maxLength={null}
                                      name="partOne"
                                      onBlur={null}
                                      onChange={[Function]}
                                      onFocus={null}
                                      onKeyPress={[Function]}
                                      path="path.to.input"
                                      placeholder={null}
                                      type="alphanum"
                                      value="JONES753116"
                                    >
                                      <input
                                        className="hd-text-input__form-control form-control"
                                        data-testid="text-input"
                                        disabled={true}
                                        id="partOne"
                                        inputMode={null}
                                        maxLength={null}
                                        name="partOne"
                                        onBlur={null}
                                        onChange={[Function]}
                                        onFocus={null}
                                        onKeyPress={[Function]}
                                        path="path.to.input"
                                        placeholder={null}
                                        type="alphanum"
                                        value="JONES753116"
                                      />
                                    </FormControl>
                                    <i
                                      className="fa fa-exclamation exclam hd-text-input__exclam"
                                    />
                                  </div>
                                </InputGroup>
                              </ForwardRef>
                            </ForwardRef>
                          </div>
                        </Col>
                        <Col
                          md={3}
                          xs={5}
                        >
                          <div
                            className="col-md-3 col-5"
                          >
                            <ForwardRef
                              className="dln-container__dln-input"
                              id="partTwo"
                              isInvalidCustom={false}
                              maxLength={5}
                              name="partTwo"
                              onBeforeOpen={null}
                              onBlur={null}
                              onCancel={null}
                              onChange={[Function]}
                              onClick={null}
                              onConfirm={null}
                              onDelete={null}
                              onEdit={null}
                              onFocus={null}
                              onKeyPress={[Function]}
                              onSelect={null}
                              reference={
                                Object {
                                  "current": <input
                                    class="hd-text-input__form-control form-control"
                                    data-testid="text-input"
                                    id="partTwo"
                                    maxlength="5"
                                    name="partTwo"
                                    path="path.to.input"
                                    type="alphanum"
                                    value=""
                                  />,
                                }
                              }
                              type="alphanum"
                              value=""
                              webAnalyticsEvent={
                                Object {
                                  "event_action": "Tell us your driving licence number and you could get a discount...",
                                  "event_value": "Please enter the next five
characters:",
                                }
                              }
                            >
                              <ForwardRef
                                allowLeadingZero={false}
                                appendContent={null}
                                appendLabel={null}
                                className="dln-container__dln-input"
                                customClassName=""
                                data={null}
                                disabled={false}
                                icon={null}
                                id="partTwo"
                                inputMode={null}
                                isInvalidCustom={false}
                                mask={null}
                                maskChar="_"
                                maxLength={5}
                                name="partTwo"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyPress={[Function]}
                                path="path.to.input"
                                placeholder={null}
                                preText={null}
                                reference={
                                  Object {
                                    "current": <input
                                      class="hd-text-input__form-control form-control"
                                      data-testid="text-input"
                                      id="partTwo"
                                      maxlength="5"
                                      name="partTwo"
                                      path="path.to.input"
                                      type="alphanum"
                                      value=""
                                    />,
                                  }
                                }
                                reset={false}
                                size={null}
                                svgIcon={null}
                                thousandSeprator={false}
                                tickIcon={false}
                                type="alphanum"
                                value=""
                              >
                                <InputGroup
                                  className="dln-container__dln-input "
                                  size={null}
                                >
                                  <div
                                    className="dln-container__dln-input  input-group"
                                  >
                                    <FormControl
                                      className="hd-text-input__form-control"
                                      data-testid="text-input"
                                      disabled={false}
                                      id="partTwo"
                                      inputMode={null}
                                      maxLength={5}
                                      name="partTwo"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      onFocus={[Function]}
                                      onKeyPress={[Function]}
                                      path="path.to.input"
                                      placeholder={null}
                                      type="alphanum"
                                      value=""
                                    >
                                      <input
                                        className="hd-text-input__form-control form-control"
                                        data-testid="text-input"
                                        disabled={false}
                                        id="partTwo"
                                        inputMode={null}
                                        maxLength={5}
                                        name="partTwo"
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onFocus={[Function]}
                                        onKeyPress={[Function]}
                                        path="path.to.input"
                                        placeholder={null}
                                        type="alphanum"
                                        value=""
                                      />
                                    </FormControl>
                                    <i
                                      className="fa fa-exclamation exclam hd-text-input__exclam"
                                    />
                                  </div>
                                </InputGroup>
                              </ForwardRef>
                            </ForwardRef>
                          </div>
                        </Col>
                        <Col
                          xs={12}
                        >
                          <div
                            className="col-12"
                          >
                            <HDAlertRefactor
                              className="margin-top-md mb-0"
                              message={null}
                            />
                          </div>
                        </Col>
                        <div
                          className="wizard-buttons-footer"
                        >
                          <Row
                            noGutters={false}
                          >
                            <div
                              className="row"
                            >
                              <Col>
                                <div
                                  className="col"
                                >
                                  <HDButtonRefactor
                                    className="hd-btn-primary button-continue"
                                    disabled={false}
                                    fullWidth={false}
                                    id="dln-continue-btn"
                                    label="Continue"
                                    onClick={[Function]}
                                    onKeyPress={null}
                                    size={null}
                                    type="button"
                                    variant="primary"
                                    webAnalyticsEvent={
                                      Object {
                                        "event_action": "Continue - Redirecting from: HastingsPersonalDetails_DrivingLicence_Number",
                                        "event_value": "Continue",
                                      }
                                    }
                                  >
                                    <Button
                                      active={false}
                                      block={false}
                                      className="hd-btn hd-btn-primary hd-btn-primary button-continue"
                                      disabled={false}
                                      id="dln-continue-btn"
                                      onClick={[Function]}
                                      onKeyPress={null}
                                      size={null}
                                      type="button"
                                      variant="primary"
                                    >
                                      <button
                                        className="hd-btn hd-btn-primary hd-btn-primary button-continue btn btn-primary"
                                        disabled={false}
                                        id="dln-continue-btn"
                                        onClick={[Function]}
                                        onKeyPress={null}
                                        type="button"
                                      >
                                        Continue
                                      </button>
                                    </Button>
                                  </HDButtonRefactor>
                                </div>
                              </Col>
                            </div>
                          </Row>
                        </div>
                      </div>
                    </Row>
                  </div>
                </Col>
              </div>
            </Row>
          </div>
        </Container>
      </HDDriverLicenceNumberPage>
    </Connect(HDDriverLicenceNumberPage)>
  </Provider>
</Component>
`;
